main_v2.py

```diff
@@ -1,109 +1,128 @@
+import os
 import requests
 import xml.etree.ElementTree as ET
-from tabulate import tabulate
 import pandas as pd
 from openpyxl import Workbook
 from openpyxl.utils.dataframe import dataframe_to_rows
 from datetime import datetime
 from tkinter import Tk, filedialog
-
-now = datetime.now()
-formatted_date = now.strftime("%Y-%m-%d-%H-%M")
-
-def consultar_awb(site_id, password, awb, historico=False):
-    url = 'https://xmlpi-ea.dhl.com/XMLShippingServlet'
-    headers = {'Content-Type': 'application/xml'}
-
-    data = f'''<?xml version="1.0" encoding="UTF-8"?>
-        <req:KnownTrackingRequest xmlns:req="http://www.dhl.com" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.dhl.com TrackingRequestKnown.xsd" schemaVersion="1.0">
-            <Request>
-                <ServiceHeader>
-                    <SiteID>{site_id}</SiteID>
-                    <Password>{password}</Password>
-                </ServiceHeader>
-            </Request>
-            <LanguageCode>BR</LanguageCode>
-            <AWBNumber>{awb}</AWBNumber>
-            <LevelOfDetails>ALL_CHECK_POINTS</LevelOfDetails>
-        </req:KnownTrackingRequest>'''
-
-    # Faz a requisição
-    response = requests.post(url, headers=headers, data=data)
-
-    # Verifica se a requisição foi bem-sucedida
-    if response.status_code == 200:
-        # Analisa a resposta XML
-        root = ET.fromstring(response.text)
-
-        # Extrai informações relevantes
+from dotenv import load_dotenv
+
+class DHLRequestBuilder:
+    def __init__(self, site_id, password, awb):
+        self.site_id = site_id
+        self.password = password
+        self.awb = awb
+        
+    def build_xml(self):
+        xml_template = """<?xml version="1.0" encoding="UTF-8"?>
+            <req:KnownTrackingRequest xmlns:req="http://www.dhl.com" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.dhl.com TrackingRequestKnown.xsd" schemaVersion="1.0">
+                <Request>
+                    <ServiceHeader>
+                        <SiteID>{}</SiteID>
+                        <Password>{}</Password>
+                    </ServiceHeader>
+                </Request>
+                <LanguageCode>BR</LanguageCode>
+                <AWBNumber>{}</AWBNumber>
+                <LevelOfDetails>ALL_CHECK_POINTS</LevelOfDetails>
+            </req:KnownTrackingRequest>"""
+        return xml_template.format(self.site_id, self.password, self.awb)
+
+class DHLTracker:
+    def __init__(self, site_id, password):
+        self.site_id = site_id
+        self.password = password
+
+    def get_awb(self, awb, exists_history=False):
+        try:
+            load_dotenv()
+            url = os.getenv("URL")
+            headers_str = os.getenv("HEADERS")
+
+            # Convertendo a string JSON em um dicionário
+            headers = eval(headers_str)
```

Comment:
aqui nao poderiamos usar o `json.loads`? vi em alguns lugares que `eval` uma má prática: https://stackoverflow.com/questions/1832940/why-is-using-eval-a-bad-practice

Reply:
De fato, trata-se de uma função obsoleta também